src/collapse/dtrace.rs

```diff
@@ -136,15 +137,21 @@ impl Folder {
         }
     }
 
-    fn remove_offset(line: &str) -> &str {
-        let mut line = line.rsplitn(2, '+');
-        // at least one element will be returned
-        let second = line.next().unwrap();
-        if let Some(first) = line.next() {
-            first
-        } else {
-            second
+    fn remove_offset(line: &str) -> (bool, bool, bool, &str) {
+        let mut could_be_java = false;
+        let mut could_be_cpp = false;
+        let mut has_colon = false;
+        let bytes = line.as_bytes();
+        for offset in 0..bytes.len() {
+            match bytes[offset] {
+                b'>' if offset > 0 && bytes[offset - 1] == b'-' => could_be_java = true,
+                b':' if offset > 0 && bytes[offset - 1] == b':' => could_be_cpp = true,
+                b';' => has_colon = true,
+                b'+' => return (could_be_java, could_be_cpp, has_colon, &line[..offset]),
```

Comment:
This is no longer the same as the old function -- you're now removing everything following the _first_ `+`, rather than the _last_. Don't know whether the two are different in practice though.

Reply:
I was wondering the same but in the examples I have couldn't find a case where a + was anywhere but at the end.

Reply:
Yeah, you're probably right that it's extremely uncommon if it ever happens. I doubt that looking through each string fully once carries all that much overhead though. Maybe we can quantify the difference? I'd _like_ to prefer the current semantics if the performance difference is minimal.

Reply:
Tested: Going to the end makes very little (way below noise) difference (as the + is late in the line anyway) so lets do it propper :) 👍